virttest/utils_misc.py

```diff
@@ -2170,24 +2170,30 @@ def _make_up_pattern(flags):
     if ARCH in ('ppc64', 'ppc64le'):
         return []     # remove -cpu and leave it on qemu to decide
 
-    cpu_types = {"AuthenticAMD": ["Opteron_G5", "Opteron_G4", "Opteron_G3",
-                                  "Opteron_G2", "Opteron_G1"],
-                 "GenuineIntel": ["Broadwell", "Broadwell-noTSX",
+    cpu_types = {"AuthenticAMD": ["EPYC", "Opteron_G5", "Opteron_G4",
+                                  "Opteron_G3", "Opteron_G2", "Opteron_G1"],
+                 "GenuineIntel": ["KnightsMill",
+                                  "Skylake-Server", "Skylake-Client",
+                                  "Broadwell", "Broadwell-noTSX",
                                   "Haswell", "Haswell-noTSX", "SandyBridge",
                                   "Westmere", "Nehalem",
                                   "Penryn", "Conroe"]}
-    cpu_type_re = {"Opteron_G5": "f16c,fma,tbm",
-                   "Opteron_G4":
-                   "avx,xsave,aes,sse4.2|sse4_2,sse4.1|sse4_1,cx16,ssse3,sse4a",
+    cpu_type_re = {"EPYC": "avx2,adx,bmi2,sha_ni",
+                   "Opteron_G5": "f16c,fma4,xop,tbm",
+                   "Opteron_G4": ("fma4,xop,avx,xsave,aes,sse4.2|sse4_2,"
+                                  "sse4.1|sse4_1,cx16,ssse3,sse4a"),
                    "Opteron_G3": "cx16,sse4a",
                    "Opteron_G2": "cx16",
                    "Opteron_G1": "",
+                   "KnightsMill": "avx512_4vnniw,avx512pf,avx512er",
```

Comment:
Suggest to remove the KnightsMill for now and add it back when it lands downstream

Reply:
@zhguo01 @yanan-fu if the certain cpu model is not supported by qemu, the current approach will skip it and then looking for the target from other candidates. So I'd suggest to keep it here for future support.

```python
def get_qemu_best_cpu_model(params):
    """
    Try to find out the best CPU model available for qemu.

    This function can't be in qemu_vm, because it is used in env_process,
    where there's no vm object available yet, and env content is synchronized
    in multi host testing.

    1) Get host CPU model
    2) Verify if host CPU model is in the list of supported qemu cpu models
    3) If so, return host CPU model
    4) If not, return the default cpu model set in params, if none defined,
        return 'qemu64'.
    """
    host_cpu_models = get_host_cpu_models()
    qemu_binary = get_qemu_binary(params)
    qemu_cpu_models = get_qemu_cpu_models(qemu_binary)
    # Let's try to find a suitable model on the qemu list
    for host_cpu_model in host_cpu_models:
        if host_cpu_model in qemu_cpu_models:     # ** here is the check
            return host_cpu_model
    # If no host cpu model can be found on qemu_cpu_models, choose the default
    return params.get("default_cpu_model", None)
```

Reply:
@luckyh  yes, it will be skipped if QEMU don't support it and check the next model. 
I just worried if there are any gap between upstream and downstream in the future. But low probability i think.
it is ok for me to keep it currently, we can update it if there are any problem when 3.0.0 be pulled to downstream.

Reply:
This is ok for me